vsm

```diff
@@ -165,6 +167,15 @@ class State(object):
                     item = item.replace(" ", "").split("=")
                     vars(self.variables)[item[0]] = item[1]
 
+        # inject this object into the globals dictionary so it will be available
+        # to the function we're executing (since it won't really be filled in
+        # until after this constructor completes)
+        global_vars = globals()
+        global_vars["state"] = self
```

Comment:
Change looks good to me, though I don't understand this bit. Is it because the conditions use a "state." prefix? If so, couldn't we just use self.variables directly instead?